src/ngraph/frontend/onnxifi/backend.hpp

```diff
@@ -0,0 +1,85 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>  // std::shared_ptr
+#include <string>  // std::string
+#include <utility> // std::move
+#include <vector>  // std::vector
+
+#include "ngraph/function.hpp"
+#include "ngraph/runtime/backend.hpp"
+#include "ngraph/runtime/tensor_view.hpp"
+
+namespace ngraph
```

Comment:
Why this class? Use runtime::Backend. Is it missing something you need?

Reply:
Yes, `runtime::Backend` is missing a log of from what ONNXIFI requires. This is just an early implementation of `onnxifi::Backend`. Consecutive PRs will add much more to the class.

Reply:
Can you please add comment to the code explaining this? Adding comments to the PR don't help people in the future.

Reply:
done